roles/nighthawk/tasks/main.yml

```diff
@@ -0,0 +1,238 @@
+
+  - include_tasks: edge-tasks.yml
+    when: "'edge' in {{ workload_args.terminations | default([]) }} or 'mix' in {{ workload_args.terminations | default([]) }}"
+
+  - include_tasks: passthrough-tasks.yml
+    when: "'passthrough' in {{ workload_args.terminations | default([]) }} or 'mix' in {{ workload_args.terminations | default([]) }}"
+
+  - include_tasks: reencrypt-tasks.yml
+    when: "'reencrypt' in {{ workload_args.terminations | default([]) }} or 'mix' in {{ workload_args.terminations | default([]) }}"
+  
+  - include_role:
+      name: benchmark_state
+      tasks_from: set_state
+    vars:
+      state: Starting Servers
+
+  when: benchmark_state.resources[0].status.state == "Building" and resource_kind == "pod"
+
+- block:
+
+  - name: Get server pods
+    k8s_facts:
+      kind: Pod
+      api_version: v1
+      namespace: '{{ operator_namespace }}'
+      label_selectors:
+        - type = {{ ansible_operator_meta.name }}-bench-nighthawk-server-{{ trunc_uuid }}
+    register: server_pods
+
+  - include_role:
+      name: benchmark_state
+      tasks_from: set_state
+    vars:
+      state: "Starting Clients"
+    when: "workload_args.terminations|default([])|length * workload_args.number_of_routes|default(1)|int == server_pods | json_query('resources[].status[]')|selectattr('phase','match','Running')|list|length"
+
+  when: benchmark_state.resources[0].status.state == "Starting Servers" and resource_kind == "pod"
+
+- block:
+
+  - name: Get server pod info
+    k8s_facts:
+      kind: Pod
+      api_version: v1
+      namespace: '{{ operator_namespace }}'
+      label_selectors:
+        - type = {{ ansible_operator_meta.name }}-bench-nighthawk-server-{{ trunc_uuid }}
+    register: server_pods
+  
+  - name: Get route info
+    k8s_facts:
+      kind: Route
+      api_version: route.openshift.io/v1
+      namespace: '{{ operator_namespace }}'
+      label_selectors:
+        - type = {{ ansible_operator_meta.name }}-bench-nighthawk-route-{{ trunc_uuid }}
+    register: routes
+
+  - block:
+    - name: Start Client(s)
+      k8s:
+        definition: "{{ lookup('template', 'workload.yml.j2') | from_yaml }}"
+      with_items: "{{ server_pods.resources }}"
+      vars:
+        routerCanonicalName: "{{ routes.resources[0] | json_query('status.ingress[0].routerCanonicalHostname') }}"
+      when: ( workload_args.serviceip|default(False) == False and server_pods.resources|length > 0 )
+    
+    - name: Start Client(s) with nodeport serviceIP
+      k8s:
+        definition: "{{ lookup('template', 'workload.yml.j2') | from_yaml }}"
+      with_items: "{{ server_pods.resources }}"
+      vars:
+        routerCanonicalName: "{{ routes.resources[0] | json_query('status.ingress[0].routerCanonicalHostname') }}"
+      when: 
+        - ( workload_args.serviceip|default(False) == True and server_pods.resources|length > 0 and 
+            workload_args.service_type | default("ClusterIP") == "NodePort" )
```

Comment:
This block can be removed, NodePort is not available anymore

Reply:
Done as we have decided to keep clusterIP as default